src/server/scripts/Kalimdor/zone_feralas.cpp

```diff
@@ -52,7 +52,42 @@ class spell_gordunni_trap : public SpellScript
     }
 };
 
+enum ZAPPED_GIANTS
+{
+    NPC_SHORE_STRIDER = 5359,
+    NPC_DEEP_STRIDER = 5360,
+    NPC_LAND_WALKER = 5357,
+    NPC_WAVE_STRIDER = 5361,
+    NPC_CLIFF_GIANT = 5358
+};
+
+class spell_transmogrify : public SpellScript
+{
+    PrepareSpellScript(spell_transmogrify);
+
+    SpellCastResult CheckTarget()
+    {
+        Unit* target = GetExplTargetUnit(); // Explicit target selected by the caster
+
+        // List of valid creature IDs
+        std::set<uint32> validCreatureIds = { NPC_SHORE_STRIDER, NPC_DEEP_STRIDER, NPC_LAND_WALKER, NPC_WAVE_STRIDER, NPC_CLIFF_GIANT };
+
+        // Validate the target
+        if (!target || validCreatureIds.find(target->GetEntry()) == validCreatureIds.end())
+            return SPELL_FAILED_BAD_TARGETS; // Prevent the spell cast with an error
+
+        return SPELL_CAST_OK; // Allow the spell cast
+    }
+
+    void Register() override
+    {
+        // Register the check cast hook
```

Comment:
also what are all these unnecessary comments

Reply:
For you they might be unnecessary. This is my very first spell script and having these comments helped me to learn and understand it better.

Reply:
yeah, if we were to add comments to all scripts like these it would be very bloated. These comments are good in the base for doxygen for example